### DISCUSSION TITLE
v3.1.5

### DISCUSSION THREAD

**Discussion by osy** [osy86/HaC-Mini#984] [Releases]
## Changes
* Fix Metal issues on macOS Monterey 12.3

## Known Issues
* LAN2 is not working on macOS Monterey (#947). This requires a driver update from Small Tree.

## Messages
* **Important**: Apple has started requiring Secure Boot for updating to macOS Monterey 12.2 and above. See the mini-guide below for more details.
* If you are upgrading from v2.17 and below, please read the notes at the top of [v3.0.1](https://github.com/osy/HaC-Mini/releases/tag/v3.0.1)
* [As announced a few months ago](https://github.com/osy/HaC-Mini/discussions/578), this project will be transitioning to a long-term support model (a few stability updates per year). Following Apple's own deprecation model, we expect the EOL of this project to be no earlier than 2023.

## Enabling Secure Boot
As of macOS Monterey 12.2, Apple has started requiring Secure Boot for OTA updates. HaC mini already supports Secure Boot but if you have not enabled it before, you need to perform some extra steps.

1. (Re)install HaC Mini and in the installer choose "Customize" and check "Boot Options -> Enable boot security".
2. After rebooting, repeatedly press the Alt/Cmd key at BIOS to toggle the OpenCore boot picker. Select the recovery partition.
3. Open Utilities -> Terminal and type the following: `bless --folder "/Volumes/Macintosh HD/System/Library/CoreServices" --bootefi --personalize` (replace "Machintosh HD" with the name of your Monterey boot volume if needed)
4. Reboot and you should be able to boot into Monterey again. If you still do not see OTA updates, open Terminal and run `sudo /System/Library/PrivateFrameworks/Seeding.framework/Versions/A/Resources/seedutil unenroll`

If these steps do not work, you may have to re-install macOS Monterey from recovery to get secure boot to work. More details about secure boot can be found [in the guide](https://osy.gitbook.io/hac-mini-guide/details/secure-boot).

**Note**: When Secure Boot is enabled, you can no longer manually modify the EFI partition (config.plist, KEXTs, etc). If you wish to make modifications, you must re-install HaC mini and uncheck the boot security option.

<hr /><em>This discussion was created from the release <a href='https://github.com/osy/HaC-Mini/releases/tag/v3.1.5'>v3.1.5</a>.</em>

**Comment by Donan123** [comment]
Thanks for the detail guide!

**Reply by anti-decoy** [reply]
I can see the OTP updates after I enable secure boot, but when I install the update 12.3, It can't reboot from Macos Installer disk, and the installation can't continue.   I have to reboot from the default disk. How to continue the installation?

**Reply by osy** [reply]
Did it error during the install? It could be due to a number of reasons... Your best bet would be to reinstall macOS from recovery.

**Reply by anti-decoy** [reply]
There is no error during the install.  
It's like that I can boot from the default disk because I run the bless command on it in recovery model, but I can't run bless command on the Macos installer disk in recovery model.

**Reply by osy** [reply]
You shouldn’t run bless or anything. If you run the OTA installer it should automatically restart into the installer.

**Reply by wonsage** [reply]
Do you have any new progress on this?
I am in the same situation.
The installer will be stuck at the beginning.

> There is no error during the install.  
> 
> It's like that I can boot from the default disk because I run the bless command on it in recovery model, but I can't run bless command on the Macos installer disk in recovery model.

**Reply by anti-decoy** [reply]
I solved it.

I found that I need reboot many times, and sometime it can continue the installation.

and now I also need to reboot many times to start the Macos because it often be blocked when the apple logo appears.

**Reply by anti-decoy** [reply]
I think it is the same problem as #988  https://github.com/osy/HaC-Mini/discussions/988

**Reply by anti-decoy** [reply]
> Do you have any new progress on this? I am in the same situation. The installer will be stuck at the beginning.
> 
> > There is no error during the install.
> > It's like that I can boot from the default disk because I run the bless command on it in recovery model, but I can't run bless command on the Macos installer disk in recovery model.

Just remove the DW1820A card.  and all is ok!

**Reply by wonsage** [reply]
@anti-decoy Thank you very much for your reply. I'm also using DW1820A, and I gonna try it tomorrow.

**Comment by 77Jony** [comment]
Intel ax210 bluetooth connot be work in macos monterey 12.3

**Comment by 77Jony** [comment]
Yep

bigsur 11.6.5 is ok




2022年4月3日 上午3:16，Maijestro ***@***.******@***.***>> 写道：



After update to Monterey 12.3.1 Bluetooth not longer work ! Is that correctly ?

**Comment by YuenJay** [comment]
v3.1.5 package intaller does not work on catalina with apple security update 2022-004(released 2022-5-16). This may be due to permission problems caused by security updates.

**Reply by Medizinmann-MD** [reply]
Ouch...can anyone confirm this behavior?

Any workarounds?
